core-plugins/src/main/java/co/cask/hydrator/plugin/batch/source/XMLRecordReader.java

```diff
@@ -77,7 +77,7 @@ public XMLRecordReader(FileSplit split, Configuration conf) throws IOException {
     XMLInputFactory factory = XMLInputFactory.newInstance();
     FSDataInputStream fdDataInputStream = fs.open(file);
```

Comment:
I think its closed when the xmlReader is closed (in the close method of the XMLRecordReader):
https://github.com/caskdata/hydrator-plugins/blob/5afa1ea0ac1e5de6f623a5adb21e7a910fa42eda/core-plugins/src/main/java/co/cask/hydrator/plugin/batch/source/XMLRecordReader.java#L105